### DISCUSSION TITLE
ui.table filter per column

### DISCUSSION THREAD

**Discussion by johancj** [zauberzeug/nicegui#1433] [Q&A]
### Description

Hi,

I would like to have filter per column on ui.table similar to what aggrid has.

How can this be done in NiceGUI? I asked in quasar discussion: https://github.com/quasarframework/quasar/discussions/16180#discussioncomment-6702485 but I still havn't figured out how to do this.

Best regards,
Johan Christian Jenssen

**Comment by falkoschindler** [comment]
Once you figure it out for QTable, it's certainly possible to do it with `ui.table`.
I think using scoped slots is a starting point.
But it's a bit too complex for me to quickly write down a solution.

**Comment by merlinarer** [comment]
Hi @falkoschindler , I move here. This feature is important table visualization, and could you please give a minimal solution with  nicegui
```
from nicegui import ui

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol'},
]
ui.table(columns=columns, rows=rows, row_key='name')

ui.run()
```
